### ISSUE TITLE
Hooking WMI IWbemClassObject::Get method from C++

### ISSUE THREAD

**Issue by abdullahtoqeer523** [EasyHook/EasyHook#317]
__declspec(dllexport) HRESULT __stdcall GetHook( LPCWSTR wszName, long lFlags, VARIANT *pVal, CIMTYPE *pType, long *plFlavor); // WMI::Get Function Prototype.

I am applying hook as
NTSTATUS result = LhInstallHook(
GetProcAddress(GetModuleHandle(TEXT("fastprox")), "?Get@CWbemObject@@UAGJPBGJPAUtagVARIANT@@PAJ2@Z"),
GetHook,
NULL,
&hHook );
ULONG ACLEntries[1] = { 0 };
LhSetExclusiveACL(ACLEntries, 1, &hHook);
the call is going to my "GetHook" function, but the parameters that I am passing through original call. are not going there. only garbage values are displaying. Need help.

**Comment by spazzarama** [comment]
You need to add an IntPtr parameter for “this” as first parameter

**Comment by abdullahtoqeer523** [comment]
Right! and also the VARIANT parameter is on 3rd and wszstring param is on 5th location.

**Comment by abdullahtoqeer523** [comment]
I tried to hook Get method in an .exe with LhSetInclusiveACL and it was all great. 

NTSTATUS result = LhInstallHook(
							GetProcAddress(GetModuleHandle(TEXT("fastprox")), "?Get@CWbemObject@@UAGJPBGJPAUtagVARIANT@@PAJ2@Z"),
							*GetHook,
							NULL,
							&hHook);
				
				ULONG ACLEntries[1] = { 0 };
				LhSetInclusiveACL(ACLEntries, 1, &hHook);

But when I am trying to hook in the same way in a DLL, it is not hooking there.

NTSTATUS result = LhInstallHook(
		GetProcAddress(GetModuleHandle(TEXT("fastprox")), "?Get@CWbemObject@@UAGJPBGJPAUtagVARIANT@@PAJ2@Z"),
		*GetHook,
		NULL,
		&hHook);

	if(SUCCEEDED(result))
		MessageBoxA(NULL, "Hook Installed", "DLLMain", MB_OK);
	else
		MessageBoxA(NULL, "Hook Not Installed", "DLLMain", MB_OK);

	ULONG ACLEntries[1] = { 0 };
	result = LhSetExclusiveACL(ACLEntries, 1, &hHook);

	if (SUCCEEDED(result))
		MessageBoxA(NULL, "Exclusive ACL", "DLLMain", MB_OK);

 Kindly guide.

**Comment by spazzarama** [comment]
Try:

`
ULONG ACLEntries[0] = { };			LhSetExclusiveACL(ACLEntries, 0, &hHook);
`
or
`
ULONG ACLEntries[1] = { 1 };			LhSetExclusiveACL(ACLEntries, 1, &hHook);
`

**Comment by abdullahtoqeer523** [comment]
(LhInstallHook) This method is failing in a dll as the message appears on DLL attach "Hook Not Installed".

**Comment by abdullahtoqeer523** [comment]
(LhInstallHook) This method is failing in a dll as the message appears on DLL attach "Hook Not Installed".

**Comment by abdullahtoqeer523** [comment]
Although all required libraries for WMI is included in DLL, but again it is failing on LhInstallHook.
